### PR TITLE
Repo re-organisation

### PR DESCRIPTION
**Pull request by mumby0168**
Re organised the repo to have the following folders
* `src` contains source code to be published as a NuGet package.
* `tests` any test projects for the source code.
* `samples` any samples of usage of any of the packages.

### CODE REVIEW

File: samples/AzureFunctionTier/AzureFunctionTier.csproj [hunk 1]
```diff
@@ -9,6 +9,7 @@
   </ItemGroup>
   <ItemGroup>
     <ProjectReference Include="..\..\Microsoft.Azure.CosmosRepository\src\Microsoft.Azure.CosmosRepository.csproj" />
+    <ProjectReference Include="..\..\src\Microsoft.Azure.CosmosRepository\Microsoft.Azure.CosmosRepository.csproj" />
```
Comment (IEvangelist) [inline samples/AzureFunctionTier/AzureFunctionTier.csproj:12]: Doesn't this replace line 11?

File: samples/ChangedFeedSamples.Shared/ChangedFeedSamples.Shared.csproj [hunk 2]
```diff
@@ -12,6 +12,7 @@
 
     <ItemGroup>
       <ProjectReference Include="..\..\Microsoft.Azure.CosmosRepository\src\Microsoft.Azure.CosmosRepository.csproj" />
+      <ProjectReference Include="..\..\src\Microsoft.Azure.CosmosRepository\Microsoft.Azure.CosmosRepository.csproj" />
```
Comment (IEvangelist) [inline samples/ChangedFeedSamples.Shared/ChangedFeedSamples.Shared.csproj:15]: Same here, line 15 should replace 14

File: samples/InMemoryWebTier/InMemoryWebTier.csproj [hunk 3]
```diff
@@ -7,6 +7,7 @@
 
     <ItemGroup>
       <ProjectReference Include="..\..\Microsoft.Azure.CosmosRepository\src\Microsoft.Azure.CosmosRepository.csproj" />
+      <ProjectReference Include="..\..\src\Microsoft.Azure.CosmosRepository\Microsoft.Azure.CosmosRepository.csproj" />
```
Comment (IEvangelist) [inline samples/InMemoryWebTier/InMemoryWebTier.csproj:10]: Same here, line 10 replaces 9

**Review by IEvangelist — APPROVED** [review]
Looks good for the most part, a few minor things

File: samples/InMemoryWebTier/InMemoryWebTier.csproj (hunk 3 above)
Comment (mumby0168) [reply samples/InMemoryWebTier/InMemoryWebTier.csproj:10]: Ah it should have done, hmm

Comment (mumby0168) [reply samples/InMemoryWebTier/InMemoryWebTier.csproj:9]: ammended

**Review by IEvangelist — APPROVED** [review]

**Pull request merged**